rke/templates/templates.go

```diff
@@ -238,7 +239,8 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.22.5-rancher2-2 <1.22.9-rancher1-1":   nginxIngressv110Rancher3,
 			">=1.22.9-rancher1-1 <1.23.0-rancher1-1":   nginxIngressv120Rancher2,
 			">=1.23.0-rancher1-1 <1.23.6-rancher1-1":   nginxIngressv110Rancher3,
-			">=1.23.6-rancher1-1":                      nginxIngressv120Rancher2,
+			">=1.23.6-rancher1-1 <1.23.16-rancher1-1":  nginxIngressv120Rancher2,
+			">=1.23.16-rancher1-1":                     nginxIngressv151Rancher2,
```

Comment:
Ranges need to be updated here as we want `nginxIngressv120Rancher2` to still be used for 1.24.0 through 1.24.9. It's fine to have `>=1.24.10-rancher1-1": nginxIngressv151Rancher2,` as the only 1.25 version we're introducing is on newer version of Ingress.

Reply:
@snasovich Updated the PR with your suggestions.

Reply:
@rayandas , thank you, but I think there is a misunderstanding here. We still want `v1.23.16-rancher1-1` to use new Ingress, I meant that we need to expand ranges in a way that only `v1.23.16-rancher1-1` (and later in 1.23.x) and `v1.24.10-rancher1-1` (and later) use 1.5.1 while previous versions stay on `nginxIngressv120Rancher2`.

Reply:
Ah! got it. I think this is the correct versioning @snasovich 
```
">=1.23.6-rancher1-1 <1.23.16-rancher1-1":  nginxIngressv120Rancher2,
">=1.23.16-rancher1-1 <1.24.0-rancher1-1":  nginxIngressv151Rancher2,
">=1.24.2-rancher1-1 <1.24.9-rancher1-1":    nginxIngressv120Rancher2,
">=1.24.10-rancher1-1":                                     nginxIngressv151Rancher2,

Reply:
Thank you, @rayandas . LGTM.